src/main/java/com/bandwidth/sdk/numbers/exception/ExceptionUtils.java

```diff
@@ -0,0 +1,53 @@
+package com.bandwidth.sdk.numbers.exception;
+
+import com.bandwidth.sdk.numbers.models.ErrorResponse;
+import com.bandwidth.sdk.numbers.models.orders.OrderResponse;
+import com.google.common.base.Joiner;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class ExceptionUtils {
+
+   private ExceptionUtils() {
+      // utility class; no instances
+   }
+
+   public static <T> CompletableFuture<T> catchAsyncClientExceptions(ThrowableSupplier<CompletableFuture<T>> supplier) {
+      try {
+         return supplier.get();
+      } catch (Throwable e) {
+         CompletableFuture<T> future = new CompletableFuture<>();
+         future.completeExceptionally(new NumbersClientException(e));
+         return future;
+      }
+   }
+
+   public static <T> T catchClientExceptions(ThrowableSupplier<T> supplier) {
+      try {
+         return supplier.get();
+      } catch (Throwable e) {
+         throw new NumbersClientException(e);
+      }
+   }
+
+   public static OrderResponse validateOrderResponse(Supplier<OrderResponse> orderResponseSupplier) {
+
+      OrderResponse orderResponse = orderResponseSupplier.get();
+
+      List<ErrorResponse> errorList = orderResponse.getErrorList();
+      if (errorList == null || errorList.size() == 1 && errorList.get(0).isOrderPendingError()) {
```

Comment:
Thoughts on returning empty errorList instead of null?

Reply:
We're returning the orderResponse in the case of a null errorList. The intention is this helper method just wraps any lambda that returns an OrderResponse and checks it for errors. If it doesn't find any, it just returns the OrderResponse. If it does, it throws an exception.